File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll.do")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    @RequestMapping("/findPage.do")
    public PageResult findPage(@RequestParam(name = "page" , defaultValue = "1") int page,
                                @RequestParam(name = "size" , defaultValue = "10") int size){
       return brandService.findPage(page, size);
    }

    @RequestMapping("/insert.do")
    public Result insert(@RequestBody TbBrand tbBrand){
        try {
            brandService.insert(tbBrand);
           return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }

    @RequestMapping("/findOne.do")
    public TbBrand findOne(long id){
        return brandService.findOne(id);
    }

    @RequestMapping("/update.do")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return  new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return  new Result(true,"修改失败");
        }
    }

    @RequestMapping("/delete.do")
    public Result delete(long[] ids){
        try {
            brandService.delete(ids);
            return  new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return  new Result(true,"删除成功");
        }

    }

    @RequestMapping("/search.do")
    public PageResult search(@RequestBody TbBrand tbBrand,
                             @RequestParam(name = "page" , defaultValue = "1") int page,
                             @RequestParam(name = "size" , defaultValue = "10") int size){
         return brandService.findPage(tbBrand,page,size);

    }

    @RequestMapping("/selectOptionList.do")
    public List<Map> selectOptionList(){
       return brandService.selectOptionList();
    }

}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController',function ($scope,$controller,brandService) {
    //继承baseController  共用$scope  不是真正意义上的继承
    $controller('baseController',{$scope:$scope});
    //查询所有
    $scope.findAll = function () {
        brandService.findAll().success(function (response) {
            $scope.list = response;
        });
    };
    //分页
    $scope.findPage = function (page,size) {
        brandService.findPage(page,size).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            })
    };
    //查找一个
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (brand) {
            $scope.brand = brand;
        })

    };
    //保存
    $scope.save = function () {
        var object = null;
        if ($scope.brand.id != null){
            object = brandService.update($scope.brand);
        }else{
            object = brandService.add($scope.brand);
        }
        object.success(
            function (response) {
                if (response.success){
                    $scope.reloadList();
                }else {
                    alert(response.message);
                }
            }
        )
    };
    //删除
    $scope.delete = function () {
        if (confirm('确定要删除吗？')) {
            brandService.delete($scope.selectIds).success(function (response) {
                if (response.success) {
                    $scope.reloadList();
                } else {
                    alert(response.message);
                }
            })
        }
    };
    //存储条件查询的条件
    $scope.searchBrand = {};
    //条件查询
    $scope.search = function (page,size) {
        brandService.search(page,size,$scope.searchBrand).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    }

});